codec/src/main/java/io/netty/handler/codec/compression/SnappyFrameDecoder.java

```diff
@@ -98,21 +97,24 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) t
 
             switch (chunkType) {
                 case STREAM_IDENTIFIER:
-                    if (chunkLength != SNAPPY.length) {
+                    if (chunkLength != SNAPPY_IDENTIFIER_LEN) {
                         throw new DecompressionException("Unexpected length of stream identifier: " + chunkLength);
                     }
 
-                    if (inSize < 4 + SNAPPY.length) {
+                    if (inSize < 4 + SNAPPY_IDENTIFIER_LEN) {
                         break;
                     }
 
-                    byte[] identifier = new byte[chunkLength];
-                    in.skipBytes(4).readBytes(identifier);
+                    in.skipBytes(4);
+                    int offset = in.readerIndex();
+                    in.skipBytes(SNAPPY_IDENTIFIER_LEN);
```

Comment:
sgtm ... the thought was that some other handler may be able to do something useful with the data